CMakeLists.txt

```
@@ -366,3 +366,10 @@ if(AUTOPEP8_FOUND)
   autopep8_add_subdirectory("${CMAKE_SOURCE_DIR}/bindings" "")
   autopep8_add_subdirectory("${CMAKE_SOURCE_DIR}/scripts" "")
 endif()
+
+# Removes every file in the build folder except the external libraries to
+# prevent redownloading external libraries
+add_custom_target(
+    deepclean
+    COMMAND python ${CMAKE_SOURCE_DIR}/scripts/developer_utils.py deepclean
```

it is better to use the python detected by Cmake compared to using the one in path to avoid surprises.

```suggestion
    COMMAND "${PYTHON_EXECUTABLE}" ${CMAKE_SOURCE_DIR}/scripts/developer_utils.py deepclean
```

---

I agree with @Felix (and the comment @Luthaf ). Since we are imposing the major version, we should also use it consistently.
On my machine, the `python` command would be `2.7`.